crates/autopilot/src/run_loop.rs

```diff
@@ -101,6 +104,39 @@ impl RunLoop {
             let block_deadline = self.current_block.borrow().number
                 + self.submission_deadline
                 + self.additional_deadline_for_rewards;
+            let mut order_executions = vec![];
+            for order in &solution.orders {
+                let auction_order = auction
+                    .orders
+                    .iter()
+                    .find(|auction_order| auction_order.metadata.uid == order.uid);
+                match auction_order {
+                    Some(auction_order) => {
+                        let executed_fee = match auction_order.metadata.class {
+                            OrderClass::Limit(LimitOrderClass { surplus_fee, .. }) => {
+                                match auction_order.data.partially_fillable {
+                                    // calculated by the solver, therefore we have to trust the
+                                    // driver here
+                                    true => {
+                                        ExecutedFee::Surplus(order.surplus_fee.unwrap_or_default())
+                                    }
+                                    // calculated by the protocol, therefore we can trust the
+                                    // auction
+                                    false => ExecutedFee::Surplus(surplus_fee.unwrap_or_default()),
+                                }
```

Comment:
I think trusting the solver is not ideal here. Also, the surplus fee can be computed from the settlement transaction, so we don't actually need to trust the driver in this case.

Reply:
> for example, when fetching full orders

👍 

> maybe when the settlement event is indexed

Yes, either this or on `Trade` events.

Reply:
Tried to add it to event indexing and it's super inconvenient. `Trade` event do not contain all necessary data to compute the fee, therefore additional async calls are needed which would slow down the event indexing code (which was the original reason why the `OnSettlementEventUpdater` was initially created). That said, I am thinking about implementing this in `OnSettlementEventUpdater` after all, but reducing the block delay after reading the https://github.com/cowprotocol/services/issues/1492

Reply:
Added the fee calculation to the `OnSettlementEventUpdater`